src/esm_runscripts/filelists.py

```diff
@@ -998,59 +1010,72 @@ def copy_files(config, filetypes, source, target):
+                            missing_files.update({file_target: file_source})
+                            continue
+                        if os.path.isfile(file_target) and filecmp.cmp(
+                            file_source, file_target
+                        ):
+                            logger.debug(
+                                f"Source and target file are identical, skipping {file_source}",
+                            )
+                            helpers.print_datetime(config)
+                            continue
+                        files_to_be_moved.append(
+                            {
                                 "movement_method": movement_method,
                                 "file_source": file_source,
                                 "file_target": file_target,
-                            })
-
-                            # To avoid overwriting in general experiment folder
-                            if skip_intermediate == True:
-                                file_target = avoid_overwriting(
-                                    config, file_source, file_target
-                                )
+                            }
+                        )
 
-                            # Execute movement
-                            movement_method(file_source, file_target)
+                        # To avoid overwriting in general experiment folder
+                        if skip_intermediate == True:
+                            file_target = avoid_overwriting(
+                                config, file_source, file_target
+                            )
 
-                            successful_files.append(file_source)
-                        except IOError:
-                            logger.error(
-                                f"Could not copy {file_source} to {file_target} for unknown reasons.",
+                        # Execute movement with or without futures (parallelization on/off)
+                        if (
+                            config["general"].get("parallel_file_movements", False)
+                            == "threads"
+                        ):
+                            future = client.submit(
+                                movement_method,
+                                config,
+                                file_source,
+                                file_target,
+                            )
+                            futures[(file_source, file_target)] = future
+                        else:
+                            futures[(file_source, file_target)] = movement_method(
+                                config,
+                                file_source,
+                                file_target,
                             )
-                            helpers.print_datetime(config)
-                            missing_files.update({file_target: file_source})
+
+    for (file_source, file_target), result in futures.items():
```

Comment:
I think this is wrong...

```suggestion
    for (file_source, file_target), future in futures.items():
```

Reply:
I think that looks better, yes. `client.submit` should give you back a `future`, not a resolved future `result`.